salt/_states/metalk8s_kubernetes.py

```diff
@@ -1887,3 +1887,34 @@ def apiservice_absent(name, **kwargs):
     }
 
     return ret
+
+
+def storageclass_present(
+    name, provisioner, reclaim_policy, volume_binding_mode, mount_options,
+    parameters, **kwargs
+):
+    ret = {'name': name,
+           'changes': {},
+           'result': False,
+           'comment': ''}
+
+    storageclass= __salt__['metalk8s_kubernetes.show_storageclass'](
+        name, **kwargs
+    )
+    if storageclass is None:
+        if __opts__['test']:
+            ret['result'] = None
+            ret['comment'] = 'The StorageClass is going to be created'
+            return ret
+
+        res = __salt__['metalk8s_kubernetes.create_storageclass'](
+            name=name, mount_options=mount_options, parameters=parameters,
+            provisioner=provisioner, reclaim_policy=reclaim_policy,
+            volume_binding_mode=volume_binding_mode, **kwargs
+        ),
+        ret['result'] = True
+        ret['changes'][name] = {'old': {}, 'new': res}
+    else:
```

Comment:
Also here we do not handle the case where storageclass exists but do not have the right "stuff".
I mean if we change the `mount_options` for example and run the state a second time we will not update it.
This is what we want ?

Reply:
Yes, we only use the `StorageClass` info when we provision the volume. If for whatever reason someone changes it later (say, change the `fsType`), there's really not much we can do!

Reply:
Yeah, changing the value of an SC isn't going to impact the existing volumes.

But if:
- in 2.4, I have a StorageClass `Foo` that defines `rw` as `mountOptions`.
- in 2.5, I have updated the StorageClass `Foo` to have `rw,some_new_stuff` as `mountOptions`.

Of course I don't want existing volume to be impacted, but we probably want new volumes to be mounted with `some_new_stuff`.
In that case, I should update the StorageClass if it exists no?

Otherwise `SC` are never "upgraded".

Reply:
Done